lib/temporal/client.rb

```diff
@@ -200,6 +201,24 @@ def query_workflow(workflow, query, workflow_id, run_id, *args, namespace: nil,
       )
     end
 
+    # Query a running workflow for its stack trace
+    #
+    # @param workflow [Temporal::Workflow, nil] workflow class or nil
+    # @param workflow_id [String]
+    # @param run_id [String]
+    # @param namespace [String, nil] if nil, choose the one declared on the workflow class or the
+    #   global default
+    # @param query_reject_condition [Symbol] check Temporal::Connection::GRPC::QUERY_REJECT_CONDITION
+    def query_workflow_stack_trace(workflow, workflow_id, run_id, namespace: nil, query_reject_condition: nil)
```

Comment:
I added the one for getting the stack trace since it is straight forward. Calling it returns a stack trace. It could be useful in a variety of circumstances, including just being invoked by tctl to debug.

The list query is not typical. It returns its result by raising an error with a specifically formatted error message. In the Go and Java SDKs, such a query doesn't even exist. They just happen to return this error when the query is unknown. temporal-web then deliberately sends a made up query to scrape the results of the error. This is much more hacky. I could catch the error here and return the list, but it's pushing the bounds of a supported API.

Reply:
@jeffschoner but it's not what's happening here — looking at the implementation this is literally just currying the `query_workflow` with a preset query name. If it's atypical then we need to provide custom handling of the result. Since this is part of a public API we need to think it through before exposing to others (changing it afterwards will be more difficult)